src/video/kmsdrm/SDL_kmsdrmvideo.c

```diff
@@ -53,19 +53,16 @@
 #include <errno.h>
 
 #ifdef __OpenBSD__
-static SDL_bool openbsd69orgreater = SDL_FALSE;
-#define KMSDRM_DRI_PATH openbsd69orgreater ? "/dev/dri/" : "/dev/"
-#define KMSDRM_DRI_PATHSIZE    openbsd69orgreater ? 9 : 5
-#define KMSDRM_DRI_DEVNAME openbsd69orgreater ? "card" : "drm"
-#define KMSDRM_DRI_DEVNAMESIZE openbsd69orgreater ? 4 : 3
+static SDL_bool moderndri = SDL_FALSE;
 #else
-#define KMSDRM_DRI_PATH "/dev/dri/"
-#define KMSDRM_DRI_PATHSIZE    9
-#define KMSDRM_DRI_DEVNAME "card"
-#define KMSDRM_DRI_DEVNAMESIZE 4
+static SDL_bool moderndri = SDL_TRUE;
 #endif
 
-#define KMSDRM_DRI_CARDPATHFMT KMSDRM_DRI_PATH KMSDRM_DRI_DEVNAME "%d"
+static char kmsdrm_dri_path[16] = "";
```

Comment:
This syntax isn't supported by all compilers, go ahead and just leave it uninitialized, which implicitly initializes to 0.